### PR TITLE
Email with custom notifications not working

### PR DESCRIPTION
**Pull request by dmzoneill**
##### SUMMARY
This targets "Email with custom notifications not working"
related #13983 

This checks for blank body and add templates errors for the user.

##### ISSUE TYPE
 - Bug, Docs Fix or other nominal change

##### COMPONENT NAME
 - API

### CODE REVIEW

File: awx/main/models/notifications.py [hunk 1]
```diff
@@ -484,14 +485,27 @@ def build_notification_message(self, nt, status):
         if msg_template:
             try:
                 msg = env.from_string(msg_template).render(**context)
-            except (TemplateSyntaxError, UndefinedError, SecurityError):
-                msg = ''
+            except (TemplateSyntaxError, UndefinedError, SecurityError) as e:
+                msg = e  # __str__
+                msg += '\n'.join(traceback.format_exception(None, e, e.__traceback__))
```
Comment (jbradberry) [inline awx/main/models/notifications.py]: This is not going to work in the way that you expect.

```
In [1]: try:
   ...:     raise Exception("foo")
   ...: except Exception as e:
   ...:     msg = e
   ...:     msg += "bar"
   ...: 
---------------------------------------------------------------------------
Exception                                 Traceback (most recent call last)
Input In [1], in <cell line: 1>()
      1 try:
----> 2     raise Exception("foo")
      3 except Exception as e:

Exception: foo

During handling of the above exception, another exception occurred:

TypeError                                 Traceback (most recent call last)
Input In [1], in <cell line: 1>()
      3 except Exception as e:
      4     msg = e
----> 5     msg += "bar"

TypeError: unsupported operand type(s) for +=: 'Exception' and 'str'
```

File: awx/main/models/notifications.py [hunk 2]
```diff
@@ -484,14 +485,27 @@ def build_notification_message(self, nt, status):
         if msg_template:
             try:
                 msg = env.from_string(msg_template).render(**context)
-            except (TemplateSyntaxError, UndefinedError, SecurityError):
-                msg = ''
+            except (TemplateSyntaxError, UndefinedError, SecurityError) as e:
+                msg = e  # __str__
```
Comment (TheRealHaoLiu) [inline awx/main/models/notifications.py]: what does `# __str__` do?

Comment (jbradberry) [reply awx/main/models/notifications.py]: Nothing.

File: awx/main/models/notifications.py (hunk 1 above)
Comment (jbradberry) [reply awx/main/models/notifications.py]: Also, using `+=` to build up strings is an anti-pattern in Python.  I'd instead suggest something like

```
msg = '\n'.join([
    str(e),
    ...whatever...
])
```

**Review by jbradberry — CHANGES_REQUESTED** [review]
The concept seems good, but the exception handling is broken as-is.

**Review by jbradberry — APPROVED** [review]

**Pull request merged**